packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/MarkersController.java

```diff
@@ -210,20 +219,96 @@ public void onAnimationUpdate(ValueAnimator animation) {
+
+      if(startCozyMarkerData != null && 
+         endCozyMarkerData != null &&
+         endCozyMarkerData.isAnimated && 
+         !isTheSameMarker){
+
+        animateMarkerTransition(markerController, newMarker, startCozyMarkerData, endCozyMarkerData);
+      }else{
+        Convert.interpretMarkerOptions(newMarker, markerController, cozyMarkerBuilder);
+      }
+    }
+  }
+
+  private void animateMarkerTransition(MarkerController markerController, Object newMarker, CozyMarkerData startMarkerData, CozyMarkerData endMarkerData){
+    final List<BitmapDescriptor> bitmapsForFrame = new ArrayList<BitmapDescriptor>();
+    final Deque<Marker> markersQueue = new ArrayDeque<Marker>();
+
+    final int totalNumberOfSimultaneousMarkers = 3;
+    final int expectedFps = 60;
+    final int framesNumber = (int) (expectedFps * ((float) markersTransitionAnimationDuration / 1000));
+
+    for(int i = 1; i <= framesNumber; i++){
+      final BitmapDescriptor markerIconFrame = BitmapDescriptorFactory.fromBitmap(cozyMarkerBuilder.buildAnimatedMarker(startMarkerData, endMarkerData, (i*1.0f)/(framesNumber*1.0f)));
+      bitmapsForFrame.add(markerIconFrame);
     }
+
+    // templateMarkerOptions to be used as template repeatedly on all frames
+    MarkerBuilder templateMarkerBuilder = new MarkerBuilder();
+    String markerId = Convert.interpretMarkerOptionsWithoutIcon(newMarker, templateMarkerBuilder);
+    MarkerOptions templateMarkerOptions = templateMarkerBuilder.build();
+    markersQueue.add(markerController.marker);
+
+    // Animation itself    
+    final Handler handler = new Handler(Looper.getMainLooper());
+    ValueAnimator transitionAnimator = ValueAnimator.ofFloat(0f, 1f);
+    transitionAnimator.setDuration(markersTransitionAnimationDuration);
+    transitionAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+        @Override
+        public void onAnimationUpdate(ValueAnimator animation) {
+            int animationIndex = (int) ((float) animation.getAnimatedValue() * framesNumber);
+
+            if(animationIndex > 0 && animationIndex <= framesNumber){
+              templateMarkerOptions.icon(bitmapsForFrame.get(animationIndex - 1));
+              Marker newMarker = googleMap.addMarker(templateMarkerOptions);
+              markersQueue.add(newMarker);
+              googleMapsMarkerIdToDartMarkerId.put(newMarker.getId(), markerId);
+            }
+
+            if(markersQueue.
+              size() > totalNumberOfSimultaneousMarkers){
+                Marker markerToRemove = markersQueue.remove();
+                googleMapsMarkerIdToDartMarkerId.remove(markerToRemove.getId());
+                markerToRemove.remove();
+            }
+
+            if((float) animation.getAnimatedValue() == 1f){
+                handler.postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                  while(markersQueue.size() > 1){
```

Comment:
```suggestion
                  while(!markersQueue.isEmpty()){
```

Reply:
Actually I want to leave one marker in the queue, so I will just leave it like that